lifecycle-management-ui/Ahk.Lifecycle.Management.Ui/Pages/Index.razor

```diff
@@ -0,0 +1,147 @@
+            <MudSwitch @bind-Checked="@noSetGrade" Color="Color.Primary">Not graded</MudSwitch>
+        </ToolBarContent>
+        <HeaderContent>
+		    <MudTh>Repository</MudTh>
+		    <MudTh>Branch</MudTh>
+		    <MudTh>Pull Request</MudTh>
+		    <MudTh>Graded</MudTh>
+	    </HeaderContent>
+	    <RowTemplate>
+		    <MudTd DataLabel="Repository">@context.Repository</MudTd>
+		    <MudTd DataLabel="Branch">
+                @if(getBranchCreate(context) == null)
+                {
+                    <MudText Color="Color.Error" Typo="Typo.body2" Class="mud-text-secondary">No new branch</MudText>
+                } else
+                {
+                    <MudText Color="Color.Success" Typo="Typo.body2" Class="mud-text-secondary">@getBranchCreate(context)?.Branch</MudText>
+                }
+            </MudTd>
+		    <MudTd DataLabel="PullRequest">
+                @if(getPullRequest(context) == null)
+                {
+                    <MudText Color="Color.Error" Typo="Typo.body2" Class="mud-text-secondary">No Pull Request</MudText>
+                } else
+                {
+                    <MudLink Href="@getPullRequest(context)?.HtmlUrl" Color="Color.Info" Typo="Typo.body2" Class="mud-text-secondary">#@getPullRequest(context)?.Number (@getPullRequest(context)?.Action)</MudLink>
+                }
+            </MudTd>
+		    <MudTd DataLabel="Graded">
+                @if(hasSetGrade(context))
+                {
+                    <MudText Color="Color.Error" Typo="Typo.body2" Class="mud-text-secondary">Not graded</MudText>
+                } else
+                {
+                    <MudText Color="Color.Success" Typo="Typo.body2" Class="mud-text-secondary">Graded</MudText>
+                }
+            </MudTd>
+	    </RowTemplate>
+        <PagerContent>
+            <MudTablePager Class="pa-4" HorizontalAlignment="HorizontalAlignment.End"/>
+        </PagerContent>
+    </MudTable>
+}
+@if(message != null)
+{
+    <MudPaper Class="pa-4 mb-4">
+        <MudText Typo="Typo.body1">@message</MudText>
+    </MudPaper>
+}
+
+@code {
+    private string? searchString;
+    private IReadOnlyCollection<Statistics>? statistics;
+    private string? message;
+    private string apiKey = "";
+    private string prefix = "";
+    private string baseUrl = "";
+    private bool noSetGrade;
+    private bool noPullRequest;
+    private bool noBranchCreate;
+
+    protected override void OnInitialized()
+    {
+        message = "Search by filling out the inputs above";
+        prefix = "";
+    }
+
+    private async Task LoadStats()
+    {
+        statistics = null;
+        message = null;
+        Http.DefaultRequestHeaders.Remove("x-functions-key");
+        Http.DefaultRequestHeaders.Add("x-functions-key", apiKey);
+
+        string url = string.Format("{0}/{1}/{2}", baseUrl.TrimEnd('/'), config["endpoint"], prefix.TrimStart('/'));
```

Comment:
A HttpClient-nek be lehet állítani egy [base address-t](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.baseaddress?msclkid=b0aa52a7cf5f11ec812effd95420c473&view=net-6.0) ott, ahol beregisztrálod. Azután itt már csak a relatív URL-re kell kiadni a kérést.

Reply:
done